assembl/lib/sqla.py

```diff
@@ -528,9 +529,13 @@ def get_instance(cls, identifier, session=None):
     def get_database_id(cls, uri):
         """Parse a URI to extract the database ID"""
         if isinstance(uri, types.StringTypes):
-            if not uri.startswith('local:') or '/' not in uri:
-                return
-            uriclsname, num = uri[6:].split('/', 1)
+            if uri.startswith('local:') and '/' in uri:
+                uriclsname, num = uri[6:].split('/', 1)
+            else:
+                try:
+                    uriclsname, num = Node.from_global_id(uri)
+                except Exception as e:
+                    return
```

Comment:
You should log the exception to raven, then return `None`. Will there be defensive logic downstream for a potential `None` db id?

Reply:
This is used all over the place, and code downstream is supposed to be defensive.
As for logging... if we knew it was either a local: type ID or a GraphQL idea or... then it would make sense to log here, but the point of this function is that it checks if your ID makes sense at all in any known scheme. The failure of one scheme (even the last scheme in line) is not information. (also, maybe there's another scheme to be tried in an overridden function, I'm thinking about it...) SO I say logging should be downstream as well. What I can do is add a comment about it.